### PR TITLE
Fix incorrect parsing of None value parameters.

### PR DESCRIPTION
**Pull request by melaniedc**
When a parameter is set as None it gets parsed as the string "None". By not setting the flag in pipeline_options.py we allow it to be set to the default value, which correctly resolves to None.

R: @robertwb 
R: @lukecwik 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ n/a ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ n/a ] Update `CHANGES.md` with noteworthy changes.
 - [ n/a ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/options/pipeline_options.py
Comment (AnandInguva) [inline sdks/python/apache_beam/options/pipeline_options.py:287]: Can you also write a small test for this `None` type args?

**Review by AnandInguva — APPROVED** [review]

File: sdks/python/apache_beam/options/pipeline_options.py
Comment (melaniedc) [reply sdks/python/apache_beam/options/pipeline_options.py:287]: Added a unit test case and confirmed it fails without and passes with my change in my local env.

**Comment by AnandInguva** [comment]
PythonDocs test is unrelated and fix is provided at https://github.com/apache/beam/pull/25000

**Review by AnandInguva — APPROVED** [review]
LGTM once formatting is fixed.

File: sdks/python/apache_beam/options/pipeline_options_test.py
Comment (AnandInguva) [inline sdks/python/apache_beam/options/pipeline_options_test.py:320]:
```suggestion
    options_dict = {'test_arg_none': None, 'test_arg_int': 5}
```

**Comment by AnandInguva** [comment]
R : @damccorm @jrmccluskey  for final review ( Valentyn on vacation)

**Review by jrmccluskey — APPROVED** [review]
LGTM, will wait on tests that required committer approval to get finished ahead of merging.

**Comment by damccorm** [comment]
Run Python PreCommit

**Pull request merged**
